### PR TITLE
Add polytopal mesh support to single-phase and multi-phase poromechanics

### PR DESCRIPTION
**Pull request by andrea-borio**
This PR enables the VEM support for single-phase and multiphase poromechanics kernels.

It adds two integrated tests:
- `inputFiles/poromechanics/PoroElastic_Mandel_prism6_smoke.xml` (single-phase, hexagonalPrisms)
- `inputFiles/poromechanics/PoroElastic_hybridHexPrism_co2_3d.xml` (multiphase, hexahedra, pentagonalPrisms, hexagonalPrisms, heptagonalPrisms, octagonalPrisms)

Additional meshes in https://github.com/GEOSX/GEOSXDATA/pull/12

The PR is ready for review and will require a rebaseline 
Rebaseline done in https://github.com/GEOSX/integratedTests/pull/286

### CODE REVIEW

**Comment by francoishamon** [comment]
@andrea-borio 

To summarize my latest modifications:

**Single-phase poromechanics:**
The PR adds a single-phase poromechanics integrated test (`inputFiles/poromechanics/PoroElastic_Mandel_prism6_smoke.xml`) that tests VEM-MFD with hexagonal prisms (now matching the analytical solution).
The PR also adds a benchmark test (`inputFiles/poromechanics/PoroElastic_Mandel_prism6_benchmark.xml`) that tests VEM with more refined hexagonal prisms meshes. I moved the more refined meshes to GEOSXDATA and converted them to vtu format.

**Multiphase poromechanics:**
The PR adds an integrated test (`inputFiles/poromechanics/PoroElastic_hybridHexPrism_co2_3d.xml`) with hexahedra, pentagonalPrisms, hexagonalPrisms, heptagonalPrisms, octagonalPrisms using the existing `pebi3d.vtu` mesh.

The main challenge is now to pass the MacOSX Travis test that has been difficult lately (the others CI tests are passing now)

**Review by francoishamon — APPROVED** [review]

**Review by andrea-borio — COMMENTED** [review]
Aside from the suggested modification to the input file, I think that the PR is ready to merge. If I understand well, we don't have a multiphase test with a polyhedral mesh and MFD. I tried modifying the new `inputFiles/poromechanics/PoroElastic_hybridHexPrism_co2_3d.xml` copying the inputs in the other new tests but when I run the test I get
```
***** ERROR
***** LOCATION: /home/andrea/cxx/geosx-development/geosx-vemPoroelastic/src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseHybridFVMKernels.hpp:1004
***** Controlling expression (should be false): true
***** Rank 0: Unknown numFacesInElem value: 7

** StackTrace of 10 frames **
Frame 0: geosx::CompositionalMultiphaseHybridFVM::precomputeData(geosx::MeshLevel&, LvArray::ArrayView<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, 1, 0, int, LvArray::ChaiBuffer> const&) 
Frame 1: geosx::FlowSolverBase::initializePostInitialConditionsPreSubGroups() 
Frame 2: geosx::CompositionalMultiphaseBase::initializePostInitialConditionsPreSubGroups() 
Frame 3: geosx::CompositionalMultiphaseHybridFVM::initializePostInitialConditionsPreSubGroups() 
Frame 4: geosx::dataRepository::Group::initializePostInitialConditions() 
Frame 5: geosx::dataRepository::Group::initializePostInitialConditions() 
Frame 6: geosx::dataRepository::Group::initializePostInitialConditions() 
Frame 7: geosx::GeosxState::applyInitialConditions() 
Frame 8: main 
Frame 9: __libc_start_main 
Frame 10: _start 
=====
```

File: inputFiles/poromechanics/PoroElastic_Mandel_prism6_base_hybrid.xml [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    </FiniteElements>
+
+    <FiniteVolume>
+      <HybridMimeticDiscretization
+        name="singlePhaseHybridMimetic"
+        innerProductType="TPFA"/>
+    </FiniteVolume>
+  </NumericalMethods>
+
+        <!-- innerProductType="beiraoDaVeigaLipnikovManzini"/>   -->
```
[the remark this answers is not in the document]
Comment (francoishamon) [reply inputFiles/poromechanics/PoroElastic_Mandel_prism6_base_hybrid.xml]: Yes, thanks for the catch, I had forgotten to put back the consistent inner product. 

I put the PR in the merge queue.

File: inputFiles/poromechanics/PoroElastic_Mandel_prism6_base_hybrid.xml [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+    <FiniteElements>
+      <FiniteElementSpace
+          name="FE1"
+          order="1"/>
+    </FiniteElements>
+
+    <FiniteVolume>
+      <HybridMimeticDiscretization
+        name="singlePhaseHybridMimetic"
+        innerProductType="TPFA"/>
```
[the remark this answers is not in the document]
Comment (francoishamon) [reply inputFiles/poromechanics/PoroElastic_Mandel_prism6_base_hybrid.xml]: Done

**Comment by francoishamon** [comment]
> Aside from the suggested modification to the input file, I think that the PR is ready to merge. If I understand well, we don't have a multiphase test with a polyhedral mesh and MFD. I tried modifying the new `inputFiles/poromechanics/PoroElastic_hybridHexPrism_co2_3d.xml` copying the inputs in the other new tests but when I run the test I get
> 
> ```
> ***** ERROR
> ***** LOCATION: /home/andrea/cxx/geosx-development/geosx-vemPoroelastic/src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseHybridFVMKernels.hpp:1004
> ***** Controlling expression (should be false): true
> ***** Rank 0: Unknown numFacesInElem value: 7
> 
> ** StackTrace of 10 frames **
> Frame 0: geosx::CompositionalMultiphaseHybridFVM::precomputeData(geosx::MeshLevel&, LvArray::ArrayView<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, 1, 0, int, LvArray::ChaiBuffer> const&) 
> Frame 1: geosx::FlowSolverBase::initializePostInitialConditionsPreSubGroups() 
> Frame 2: geosx::CompositionalMultiphaseBase::initializePostInitialConditionsPreSubGroups() 
> Frame 3: geosx::CompositionalMultiphaseHybridFVM::initializePostInitialConditionsPreSubGroups() 
> Frame 4: geosx::dataRepository::Group::initializePostInitialConditions() 
> Frame 5: geosx::dataRepository::Group::initializePostInitialConditions() 
> Frame 6: geosx::dataRepository::Group::initializePostInitialConditions() 
> Frame 7: geosx::GeosxState::applyInitialConditions() 
> Frame 8: main 
> Frame 9: __libc_start_main 
> Frame 10: _start 
> =====
> ```
> 
> Maybe we can add a multiphase test with VEM + MFD in another PR. It was just to let you know.

Yes, you are right, the polyhedral cells are not enabled for the multiphase case yet. 

I just need to do what I had done for single-phase MFD in PR #2095 for the multiphase case: changing the dispatch to allow polyhedral cells, changing the style of the kernel to bring it up to speed with the rest of the code (add a kernel class, etc). That should not be difficult, I will do it soon.

**Review by castelletto1 — APPROVED** [review]

**Pull request merged**